### ISSUE TITLE
Better geography based layout where country codes are used

### ISSUE THREAD

**Issue by jhollway** [snlab-ch/migraph#201]

**Comment by BBieri** [comment]
I implemented a first working version of such graphs in the latest development version of `{migraph}`. Happy to get any feedback @jaeltan, @jhollway, @henriquesposito. Below is a mini reprex that runs after you update to the latest development version of `{migraph}`.

``` {r}
# Generate the unimodal network of agreements signed in 2010
membership <- manyenviron::memberships$ECOLEX_MEM %>%
   dplyr::mutate(year = stringr::str_extract(many_ID, "[:digit:]{4}")) %>%
   dplyr::filter(year == "2010") %>%
   dplyr::select(treaty_ID, CountryID, Title) %>%
   tidyr::drop_na() %>%
   migraph::as_tidygraph() %>%
   migraph::project_cols()

 # Plot the network at the specified date
 # Light theme
 network_map(membership, date, theme = "light") +
   labs(title = "International Environmental Treaties 2010",
        subtitle = "Ecolex data",
        caption = "Created with love by {migraph}")
 ggsave("light.png")
 # Dark theme
 network_map(membership, date, theme = "dark") +
   labs(title = "International Environmental Treaties 2010",
        subtitle = "Ecolex data",
        caption = "Created with love by {migraph}")
 ggsave("dark.png")
 # Earth theme
 network_map(membership, date, theme = "earth") +
   labs(title = "International Environmental Treaties 2010",
        subtitle = "Ecolex data",
        caption = "Created with love by {migraph}")
 ggsave("earth.png")
```
![light.png](https://images.zenhubusercontent.com/5fb64217ad906a65de7927e7/8fae59e2-7510-4a79-83a3-cafae54404fa)
![dark.png](https://images.zenhubusercontent.com/5fb64217ad906a65de7927e7/f08d7034-fd47-45b5-b43e-0684160a78a7)
![earth.png](https://images.zenhubusercontent.com/5fb64217ad906a65de7927e7/94053e31-8e5d-4624-848d-3ae3f4b281e9)

**Comment by BBieri** [comment]
One of the issues I am still facing is the fact that the `network_map()` function leverages `{cshapes}` data to plot historical maps with the coordinates of historical capitals and the shapes of historical countries. Plus, the function uses the `{countrycode}` package to convert COW_ID's used by `{cshapes}` to ISO3c used by `{manyenviron}` in this case. This means that we are adding two dependencies to get the total to 22 :/

Hence my suggestion of separating `{migraph}` in two distinct packages. {migraph} would retain all the manipulation and analysis functions while {migraph_viz} (or some other name) would harbor all the visualization tools for networks. This would follow the separation logic of `{tidygraph}` and `{ggraph}`. What do you think of this @jhollway ?

**Comment by henriquesposito** [comment]
This looks really good @BBieri ! 

Perhaps you can add a helper function in `{migraph}` like the `depends()` in maypkgs (or we can move `depends()` to `{manydata}` which will soon be on CRAN) to load/install packages necessary for only one other function when that specific function is called, as is the case of `{countrycode}` and `{cshapes}`. This would allow to move those packages to suggests instead of imports in description file. Just a small suggestion, thank you.

**Comment by jhollway** [comment]
Why are we using `{countrycode}` instead of `{manystates}` -- just because it's on CRAN?

**Comment by BBieri** [comment]
Exactly @jhollway . 

Thanks for the feedback @henriquesposito !  This would indeed solve my issues, however, I am not entirely sure that CRAN would like such a function bypassing their 20 dependency limits.

**Comment by jhollway** [comment]
Would it be possible to get `{manystates}` on CRAN too?

**Comment by BBieri** [comment]
Not as is @jhollway, since `{manystates}` depends on `{manypkgs}` and `{vdemdata}`  which are not and will not be on CRAN anytime soon :/

**Comment by BBieri** [comment]
However, it would be the package with the widest purpose and easiest port to CRAN out of all the data packages we have at the moment. I'll check whether we can solve these dependency issues by the beginning of next week!

**Comment by jhollway** [comment]
Great, let us know the feasibility of this.

**Comment by BBieri** [comment]
I removed the `{manydata}` dependency since the only function we were using from that package in the `/R` folder scripts was `transmutate()`. To remove the `{manypkgs}` and the `{vdemdata}` dependencies we would have to:

- Either find a way to put `{manypkgs}` on CRAN since it is used in the R preparation scripts for `{cshapes}` and `{vdemdata}`  as well as putting `{vdemdata}` in the `/data-raw` scripts.
- Or switch the `{cshapes}` and `{vdemdata}` to standard preparation scripts in `/data-raw` instead of them being functions. This has the downside of having to periodically check whether there are updated versions of the datasets and reimport them.

Any preference for either solution?

**Comment by jhollway** [comment]
Hmm, how large are the `{cshapes}` and `{vdemdata}` contributions to the package? We also need to think about keeping this lightweight/lite.

**Comment by BBieri** [comment]
- `{cshapes}` is quite large since it contains shapes of countries which is why I implemented it as a function and not a standard dataset in `/data-raw`. I think `{cshapes}` is a neat addition to our package as it allows for quick mapmaking of consistent historical geographical networks e.g. the ones I created above for example without having to make the data consistent with our "many-verse" every time.

- `{vdemdata}` is smaller and could be integrated into the `/data-raw` folder.

The bottom line with respect to package size is that importing data with an R script saves space but adds dependencies and storing data in `/raw-data` creates a heavier package with less dependencies.

**Comment by jhollway** [comment]
How much larger does cshapes make the package?

**Comment by BBieri** [comment]
For now, it does not make the package heavier since we import the data from their package via an R function. However, putting `{cshapes}` in the `/data-raw` would lead to making the package too heavy for CRAN since the entire `{cshapes}` dataset is about 25 MB.

**Comment by jhollway** [comment]
What about not having it as an `Imports` but a `Suggests` and include a `require()` in that function?
